File: template/ocaml/test/test_aoc.ml
open Alcotest

let test_part_one expected input =
  fun () -> check int "solves part one" expected (Aoc.solve_part_one input)
;;

let test_part_two expected input =
  fun () -> check int "solves part two" expected (Aoc.solve_part_two input)
;;

let () =
  run
    "AoC"
    [ "Part One", [ test_case "Example 1" `Quick (test_part_one 0 "") ]
    ; "Part Two", [ test_case "Example 1" `Quick (test_part_two 0 "") ]
    ]
;;

File: template/python/tests/test_aoc.py
import unittest
from unittest import TestCase

from main import solve_part_one, solve_part_two


class TestAoc(TestCase):
    def test_part_one(self):
        input = ""
        expected = 0

        actual = solve_part_one(input)

        self.assertEqual(expected, actual, f"failed for input {input}")

    def test_part_two(self):
        input = ""
        expected = 0

        actual = solve_part_two(input)

        self.assertEqual(expected, actual, f"failed for input {input}")


if __name__ == "__main__":
    unittest.main()
